main/webapp/modules/core/scripts/index/default-importing-controller/controller.js

```diff
@@ -265,10 +265,13 @@ Refine.DefaultImportingController.prototype.getPreviewData = function(callback,
           "limit" : numRows || 100 // More than we parse for preview anyway
         }),
         null,
-        function(data) {
-          result.rowModel = data;
-          callback(result);
-        },
+		  function(data) {
+			  if (data["rows"].length == 0) {
```

Comment:
How about checking for a few rows, but empty columns? Is that a possible scenario?

Reply:
I will have to try and see if its possible

Reply:
I think to check for a few rows there has to be some rows in the received data i.e `data["rows"].length > 0` which implies there was no parse error.

Reply:
@WaltonG : I think we could refuse to import a table that is 1x1. Anything above that is fine. But at least one row and one column. 

Even 1x1, we could accept, but no user would do that. And the solution would be to add an return at the end of the file.

It will be a good improvement from 1x5 minimum ! ;-)

Reply:
> @WaltonG : I think we could refuse to import a table that is 1x1. Anything above that is fine. But at least one row and one column.
> 
> Even 1x1, we could accept, but no user would do that. And the solution would be to add an return at the end of the file.
> 
> It will be a good improvement from 1x5 minimum ! ;-)

This would mean by-passing  the 1x5 minimum check in the backend. Would that be in order ?

Reply:
> @antoine2711: This would mean by-passing the 1x5 minimum check in the backend. Would that be in order ?

Have you investigated it?!

Reply:
The usage found so far is when parsing XML :
https://github.com/OpenRefine/OpenRefine/blob/8e5e06a4b11843bbfcb5d1db128c08aef46d2b5c/main/src/com/google/refine/importers/tree/XmlImportUtilities.java#L122-L130

Reply:
> The usage found so far is when parsing XML (...)

Could you explain to me why there is this limitation? And should it also apply to JSON?!

https://github.com/OpenRefine/OpenRefine/blob/8e5e06a4b11843bbfcb5d1db128c08aef46d2b5c/main/src/com/google/refine/importers/tree/XmlImportUtilities.java#L161

And also, is the 5 rows limit coming from this?

Regards, Antoine

Reply:
No the limit does not apply to JSON however I didnt get any iplementation of the check in the code base and on further analysis I managed to create a project from an element with 1 record

```
<wb:data xmlns:wb="http://www.worldbank.org" >
        <wb:indicator id="SP.POP.TOTL">Population, total</wb:indicator>
</wb:data>
```
![image](https://user-images.githubusercontent.com/52596583/165050776-20d5d7b1-4a3e-4f17-97d7-9b9dd6a049d4.png)

Could it have been fixed in one of the releases  ?
>From [dfhu...@gmail.com](mailto:dfhu...@gmail.com) on May 23, 2011 13:55:50:
I think this problem occurs because there isn't enough content in the file for Refine to guess where the record elements are. The next version of Refine will require you to pick the record elements yourself and hopefully won't have this problem.

Reply:
No the limit does not apply to JSON however I didnt get any iplementation of the check in the code base and on further analysis I managed to create a project from an element with 1 record

```
<wb:data xmlns:wb="http://www.worldbank.org" >
        <wb:indicator id="SP.POP.TOTL">Population, total</wb:indicator>
</wb:data>
```
![image](https://user-images.githubusercontent.com/52596583/165050776-20d5d7b1-4a3e-4f17-97d7-9b9dd6a049d4.png)

Could it have been fixed in one of the releases  ?
>From [dfhu...@gmail.com](mailto:dfhu...@gmail.com) on May 23, 2011 13:55:50:
I think this problem occurs because there isn't enough content in the file for Refine to guess where the record elements are. The next version of Refine will require you to pick the record elements yourself and hopefully won't have this problem.

Reply:
> > From [dfhu...@gmail.com](mailto:dfhu...@gmail.com) on May 23, 2011 13:55:50:
> > I think this problem occurs because there isn't enough content in the file for Refine to guess where the record elements are. The next version of Refine will require you to pick the record elements yourself and hopefully won't have this problem.

That is great! I guess you have the honor to do the fixe, since we choose the record since a few version. So, we do not this constraint any longer.

Regards, Antoine